ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -1235,14 +1262,15 @@ RouterErrorCode processServerError(ServerErrorCode errorCode) {
 
     /**
      * Process a metadata blob to find the data chunks that need to be fetched.
+     * @param responseInfo the response received for a request sent out on behalf of this chunk.
      * @param blobData the metadata blob's data.
      * @param userMetadata userMetadata of the blob
      * @param encryptionKey blob encryption key. Could be null for un-encrypted blobs
      * @throws IOException
      * @throws MessageFormatException
      */
-    private void handleMetadataBlob(BlobData blobData, byte[] userMetadata, ByteBuffer encryptionKey)
-        throws IOException, MessageFormatException {
+    private void handleMetadataBlob(ResponseInfo responseInfo, BlobData blobData, byte[] userMetadata,
```

Comment:
I noticed that responseInfo is passed into this method but not used.

Reply:
Will update.